tests/components/sensor/test_template.py

```diff
@@ -90,7 +86,7 @@ def test_invalid_name_does_not_create(self):
 
     def test_invalid_sensor_does_not_create(self):
         """Test invalid sensor."""
-        assert sensor.setup(self.hass, {
+        assert not bootstrap.setup_component(self.hass, 'sensor', {
```

Comment:
Unlike the previous setup (which always returned true) - bootstrap does seem to return false in these cases - but the next line also does the check you suggest.